tiny_dnn/layers/average_pooling_layer.h

```diff
@@ -169,7 +169,8 @@ class average_pooling_layer : public partial_connected_layer<Activation> {
                           serial_size_t in_channels,
                           serial_size_t pool_size,
                           serial_size_t stride)
-        : average_pooling_layer(in_width, in_height, in_channels, pool_size, pool_size, stride, stride, padding::valid)
+        : average_pooling_layer(in_width, in_height, in_channels, pool_size,
+                                (in_height == 1 ? 1 : pool_size), stride, stride, padding::valid)
     {}
```

Comment:
It would be much easier to read with some comments.
```cpp
        : average_pooling_layer(in_width, in_height, in_channels,
                                pool_size, // pool_size_x
                                (in_height == 1 ? 1 : pool_size), // pool_size_y
                                stride, // stride_x
                                stride, // stride_y
                                padding::valid)
```